app/components/item-thumbnail-view.js

```diff
@@ -1,45 +1,101 @@
-import { bind } from "@ember/runloop";
-import Component from "@ember/component";
+import ImagePreview from "./image-preview";
+import { getWithDefault, computed } from "@ember/object";
 import { inject as service } from "@ember/service";
 import cloudinaryImage from "../mixins/cloudinary_image";
+import _ from "lodash";
 
-export default Component.extend(cloudinaryImage, {
-  loading: true,
+export default ImagePreview.extend(cloudinaryImage, {
   store: service(),
 
-  changedSrc: function() {
-    if (!this.is_cached(this.get("src"))) {
-      this.set("loading", true);
+  initializeLightgallery() {
+    var gallery = $(`#lightGallery${this.get("item.id")}`).data("lightGallery");
+    if (gallery) {
+      gallery.destroy();
     }
-  }.observes("src"),
 
-  didReceiveAttrs() {
-    this._super(...arguments);
-    this.set("cloudinaryId", this.get("cloudinaryId"));
-    this.set("src", this.generateUrl(500, 500, true));
-    const pkgTypeType = this.get("store").peekRecord(
-      "package_type",
-      this.get("item").attributes.package_type_id
+    var lightGalleryObj = $(`#lightGallery${this.get("item.id")}`).lightGallery(
+      {
+        mode: "lg-slide",
+        zoom: true,
+        download: false,
+        scale: 1,
+        hideControlOnEnd: true,
+        closable: true,
+        loop: true,
+        counter: true,
+        enableTouch: true,
+        enableDrag: true,
+        selector: ".imageZoom"
+      }
     );
-    this.set("pkgTypeType", pkgTypeType);
+    this.set("lightGalleryObj", lightGalleryObj);
   },
 
-  onLoad: function() {
-    this.set("loading", false);
+  safeGet(model, id) {
```

Comment:
This is nice and handy function. Might be useful in many places

You can create a mixin for it and making it more configurable

```javascript
export default Ember.Mixin.create({
   safeGet: function(model, id, attr,defaultValue="N/A") {
       if(!attr) {
         throw new Error('Attribute cannot be empty');
       }
       
       if(!defaultValue) {
          throw new Error('Default value cannot be empty')
       }
       
       const record = this.get("store").peekRecord(model, id);
       return getWithDefault(record, "name", attr, defaultValue);
   }
})
```

Reply:
Thanks...i have added the mixin for it